azure-kusto-data/tests/test_functional.py

```diff
@@ -229,13 +229,12 @@ def test_column_dont_exist(self):
         response = _KustoResponseDataSetV2(json.loads(RESPONSE_TEXT))
         row = response.primary_results[0]
         self.assertRaises(IndexError, row.__getitem__, 10)
-        self.assertRaises(KeyError, row.__getitem__, "NonexistentColumn")
+        self.assertRaises(LookupError, row.__getitem__, "NonexistentColumn")
 
     def test_iterating_after_end(self):
         """Tests StopIteration is raised when the response ends."""
         response = _KustoResponseDataSetV2(json.loads(RESPONSE_TEXT))
-        iterator = response.primary_results
-        iterator.__next__()
-        iterator.__next__()
-        iterator.__next__()
-        self.assertRaises(StopIteration, iterator.__next__)
+        row_count = 0
+        for _ in response.primary_results:
+            row_count += 1
```

Comment:
Some alternatives you could use here if you want:

```python
row_count = len(list(response.primary_results))
row_count = sum(1 for _ in response.primary_results)
```

Reply:
Fixed.